client/db/src/lib.rs

```diff
@@ -344,27 +354,27 @@ pub enum DatabaseSettingsSrc {
 	Custom(Arc<dyn Database<DbHash>>),
 }
 
-impl DatabaseSettingsSrc {
+impl DatabaseSource {
 	/// Return dabase path for databases that are on the disk.
 	pub fn path(&self) -> Option<&Path> {
 		match self {
-			DatabaseSettingsSrc::RocksDb { path, .. } => Some(path.as_path()),
-			DatabaseSettingsSrc::ParityDb { path, .. } => Some(path.as_path()),
-			DatabaseSettingsSrc::Custom(_) => None,
+			DatabaseSource::Auto { .. } => {
+				// TODO:
+				unimplemented!();
```

Comment:
I'm not sure what to do here, since `DatabaseSource::Auto` has 2 paths (separate for paritydb and rocksdb).

Reply:
IIUC this is needed for polkadot to create its own dbs, so until it cannot use parity db I would say rocksdb, but later parity-db.
In fact I am not sure it is a good approach, maybe returning the base common path could be more helpful, then we could have a polkadot having its own 'Auto' logic. But then we would need a companion to this PR to append 'db' to the common path.